app/lib/actions/foreman_ansible/play_host_roles.rb

```diff
@@ -0,0 +1,56 @@
+module Actions
+  module ForemanAnsible
+    # Actions that initiaztes the playbook run for roles assigned to
+    # the host. It doest that either locally or via a proxy when available.
+    class PlayHostRoles < Actions::EntryAction
+      include ::Actions::Helpers::WithContinuousOutput
+      include ::Actions::Helpers::WithDelegatedAction
+
+      def plan(host, proxy_selector = ::ForemanAnsible::ProxySelector.new)
+        input[:host] = { :id => host.id, :name => host.fqdn }
+        proxy = proxy_selector.determine_proxy(host)
```

Comment:
In any case it should fail in `plan_delegated_action`, not while determining the proxy